modules/repository-datastax/src/main/java/com/intuit/wasabi/repository/cassandra/impl/CassandraExperimentRepository.java

```diff
@@ -333,16 +333,24 @@ protected Experiment internalGetExperiment(Application.Name appName,
         Experiment experiment = null;
 
         try {
-            ExperimentByAppNameLabel experimentAppLabel = experimentLabelIndexAccessor
-                    .getExperimentBy(appName.toString(),
-                            experimentLabel.toString()).one();
```

Comment:
If I remember correctly, then value fetched in this method are cached by wasabi. Did you notice `one()`? That means, it will fetch only 1 experiment label if there are 2.